src/utils/blockchain/math-helpers.ts

```diff
@@ -77,9 +84,28 @@ export const toWei = (amount: any, unit: Unit = 'ether') => {
       throw new Error('Unsupported unit (custom fromWei helper)');
   }
 
-  return roundToSmaller(bignumber(amount || '0').mul(10 ** decimals), 0);
+  return roundToSmaller(
+    bignumber(toValidNumberishValue(amount)).mul(10 ** decimals),
+    0,
+  );
 };
 
 export const trimZero = (amount: string) => {
   return amount.replace(/([0-9]+(\.[0-9]+[1-9])?)(\.?0+$)/, '$1');
 };
+
+export const isValidNumerishValue = (value: any) => {
+  if (
+    value === Infinity ||
+    Number(value) <= 0 ||
+    isNaN(Number(value)) ||
+    value === undefined ||
+    value === null
+  ) {
+    return false;
+  }
+  return true;
+};
+
+export const toValidNumberishValue = (value: any): string =>
+  isValidNumerishValue(value) ? bignumber(value).toString() : '0';
```

Comment:
The is valid function isn't needed. Changed the function name to be readable. 

```suggestion
export const toValidNumberString = (value: any): string =>
  value && Number.isFinite(Number(value)) ? bignumber(value).toString() : '0';
```